pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/DataDictionaryExporter.java

```diff
@@ -0,0 +1,293 @@
+                activityRepeatNum,
+                questionRepeatNum,
+                null,
+                null,
+                null) + "]]";
+        String descriptionText = (String) filterConfig.getQuestionDef().get(ESObjectConstants.QUESTION_TEXT);
+        addRowToSheet(questionRootName, "text", "MULTISELECT", descriptionText, null);
+    }
+
+    protected void addRegularVariableRow(FilterExportConfig filterConfig,
+                                             int activityRepeatNum,
+                                             int questionRepeatNum,
+                                             Map<String, Object> option,
+                                             String detailName,
+                                             FilterExportConfig parentConfig) {
+        String variableName = getColumnName(filterConfig,
+                activityRepeatNum,
+                questionRepeatNum,
+                option,
+                detailName,
+                parentConfig);
+
+        String questionType = filterConfig.getQuestionType();
+        if (questionType == null) {
+            questionType = "";
+        }
+        String dataType = QUESTION_TYPE_TO_DATA_TYPE.getOrDefault(questionType, "text");
+
+        String descriptionText = filterConfig.getColumn().getDisplay();
+        if (filterConfig.getQuestionDef() != null) {
+            String questionText = (String) filterConfig.getQuestionDef().get(ESObjectConstants.QUESTION_TEXT);
+            descriptionText = StringUtils.isNotBlank(questionText) ? questionText : descriptionText;
+        }
+        if (descriptionText.contains("_")) {
+            descriptionText = Arrays.stream(descriptionText.split("_"))
+                    .map(word -> StringUtils.toRootLowerCase(word)).collect(Collectors.joining(" "));
+        }
+        if (filterConfig.getMaxRepeats() > 1) {
+            descriptionText += "\n May have up to " + filterConfig.getMaxRepeats() +
+                    " response variables, denoted with _2, _3, etc. for each response after the first.";
+        }
+
+        String optionText = null;
+        if (filterConfig.getOptions() != null && filterConfig.getCollationSuffix() == null) {
+            optionText = filterConfig.getOptions().stream().map(opt ->
+                    opt.get(ESObjectConstants.OPTION_STABLE_ID) + " - " + opt.get(ESObjectConstants.OPTION_TEXT)
+            ).collect(Collectors.joining("\n"));
+        }
+        addRowToSheet(variableName, dataType, questionType, descriptionText, optionText);
+
+    }
+
+    protected void addAdditionalDetailRow(FilterExportConfig filterConfig,
+                                          int activityRepeatNum,
+                                          int questionRepeatNum,
+                                          Map<String, Object> option,
+                                          String detailName,
+                                          FilterExportConfig parentConfig) {
```

Comment:
I would say that such parameters list is data clump, because they are used in different methods exactly the same number of arguments with the same types, I would use parameter object design pattern: https://github.com/iluwatar/java-design-patterns/tree/master/parameter-object

Reply:
Yeah, I could see that going either way.  Here, not encasing the parameters in an object was very useful to me as I was writing the code, because then I could see which parameters I was/wasn't using in the method logic

Reply:
Yes, but having such many parameters in a method is not considered as a good technique, later when someone will touch this code for further changes he/she will probably add method which will be needed to called from this particular method and pass every parameter and it will become huge mess